src/aliceVision/fuseCut/DelaunayGraphCut.cpp

```diff
@@ -867,71 +917,113 @@ void DelaunayGraphCut::fuseFromDepthMaps(const StaticVector<int>& cams, const Po
-        double pixSizeMarginCoef = 2.0;
-        // while more points than the max points (with a limit to 10 iterations).
-        for(int filteringIt = 0; filteringIt < 10; ++filteringIt)
-        {
-            filterByPixSize(verticesCoordsPrepare, pixSizePrepare, pixSizeMarginCoef, simScorePrepare);
+    ALICEVISION_LOG_INFO("Filter initial 3D points by pixel size to remove duplicates.");
 
-            // remove points if pixSize == -1
-            removeInvalidPoints(verticesCoordsPrepare, pixSizePrepare, simScorePrepare);
+    filterByPixSize(verticesCoordsPrepare, pixSizePrepare, params.pixSizeMarginInitCoef, simScorePrepare);
+    // remove points if pixSize == -1
+    removeInvalidPoints(verticesCoordsPrepare, pixSizePrepare, simScorePrepare);
 
-            if(verticesCoordsPrepare.size() < maxVertices)
-            {
-                ALICEVISION_LOG_INFO("The number of points is below the max number of vertices.");
-                break;
-            }
-            else
-            {
-                pixSizeMarginCoef *= 2.0;
-                ALICEVISION_LOG_INFO("Increase pixel size margin coef to " << pixSizeMarginCoef << ", nb points: " << verticesCoordsPrepare.size() << ", maxVertices: " << maxVertices);
-            }
-        }
-        ALICEVISION_LOG_INFO("3D points loaded and filtered to " << verticesCoordsPrepare.size() << " points (maxVertices is " << maxVertices << ").");
-    }
+    ALICEVISION_LOG_INFO("3D points loaded and filtered to " << verticesCoordsPrepare.size() << " points.");
 
-    ALICEVISION_LOG_INFO("Create visibilities");
+    ALICEVISION_LOG_INFO("Init visibilities to compute angle scores");
     std::vector<GC_vertexInfo> verticesAttrPrepare(verticesCoordsPrepare.size());
 
-    // Initialize the vertice attributes and declare the visibility information
+    // Compute the vertices positions and simScore from all input depthMap/simMap images,
+    // and declare the visibility information (the cameras indexes seeing the vertex).
     createVerticesWithVisibilities(cams, verticesCoordsPrepare, pixSizePrepare, simScorePrepare,
-                                   verticesAttrPrepare, mp);
+                                   verticesAttrPrepare, mp, params.simFactor, params.voteMarginFactor, params.contributeMarginFactor, params.simGaussianSize);
 
-    // Export points with enough visibility support
+    ALICEVISION_LOG_INFO("Compute max angle per point");
+
+    ALICEVISION_LOG_INFO("angleFactor: " << params.angleFactor);
+    // Compute max visibility angle per point
+    // and weight simScore with angular score
+    double stat_minAngle = std::numeric_limits<double>::max(), stat_maxAngle = 0.0;
+    double stat_minAngleScore = std::numeric_limits<double>::max(), stat_maxAngleScore = 0.0;
+    #pragma omp parallel for reduction(max: stat_maxAngle,stat_maxAngleScore) reduction(min: stat_minAngle,stat_minAngleScore)
```

Comment:
`reduction(max/min:)` not available within MSVC (OpenMP 2.0). We might need a `critical` fallback in this case.

Reply:
Arf, I just use it here to log some statistics, so it's not required for the computation. We can add ifdef on the version of openmp to skip it.